api/envoy/config/wasm/v2/wasm.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package envoy.config.wasm.v2;
+
+option java_outer_classname = "WasmProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.config.wasm.v2";
+
+import "envoy/api/v2/core/base.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Wasm service]
+
+// Configuration for a Wasm VM.
+// [#next-free-field: 6]
+message VmConfig {
+  // An ID which will be used along with a hash of the wasm code (or null_vm_id) to determine which
+  // VM will be used for the plugin. All plugins which use the same vm_id and code will use the same
+  // VM. May be left blank.
+  string vm_id = 1;
+
+  // The Wasm runtime type (see source/extensions/common/wasm/well_known_names.h).
+  string runtime = 2;
+
+  // The Wasm code that Envoy will execute.
+  api.v2.core.AsyncDataSource code = 3;
+
+  // The Wasm configuration string used on initialization of a new VM (proxy_onStart).
+  string configuration = 4;
+
+  // Allow the wasm file to include pre-compiled code on VMs which support it.
+  bool allow_precompiled = 5;
+}
+
+// Base Configuration for Wasm Plugins e.g. filters and services.
```

Comment:
What is a service in this context?

Reply:
In this case it is a WASM VM which runs independently of a Filter or streams.  It is for background activities (e.g. timer or shared queue based) like stat or log collation.   

Perhaps a better name in addition to some more comments and/or links to other documentation.

Reply:
Yeah, I think we need a lexicon or something. Like, there is a lot of terminology, and we want readers to be clear on it and ensure it aligns with the rest of the Envoy world.

Reply:
@PiotrSikora let's add a lexicon into the docs.

Reply:
Moves this into the architecture doc which will then be converted to dev documentation (issue https://github.com/envoyproxy/envoy/issues/9465).